sys/malloc_thread_safe/malloc_wrappers.c

```diff
@@ -45,28 +63,53 @@ void __attribute__((used)) __wrap_free(void *ptr)
 
 void * __attribute__((used)) __wrap_calloc(size_t nmemb, size_t size)
 {
+    uinttxtptr_t pc;
+    if (IS_USED(MODULE_MALLOC_TRACING)) {
+        pc = cpu_get_caller_pc();
+    }
     /* some c libs don't perform proper overflow check (e.g. newlib < 4.0.0). Hence, we
      * just implement calloc on top of malloc ourselves. In addition to ensuring proper
      * overflow checks, this likely saves a bit of ROM */
     size_t total_size;
     if (__builtin_mul_overflow(nmemb, size, &total_size)) {
+        if (IS_USED(MODULE_MALLOC_TRACING)) {
+            printf("calloc(%u, %u) @0x%" PRIxTXTPTR " overflowed\n",
+                   (unsigned)nmemb, (unsigned)size, pc);
+        }
         return NULL;
     }
 
-    void *res = __wrap_malloc(total_size);
+    mutex_lock(&_lock);
+    void *res = __real_malloc(total_size);
+    mutex_unlock(&_lock);
```

Comment:
Is this change actually related to the PR?

Reply:
Yes, otherwise you get two prints for one dyn alloc call